### PR TITLE
Simple reader

### PR DESCRIPTION
**Pull request by xiaodaigh**
A simple reader that returns a table. Nothing more nothing less. Just simple API.

```
# `tbl` is a Tables.jl compatible table
tbl = read_parquet(path_to_parquet_file)

# example for how to convertconvert to DataFrame
using DataFrames
df = DataFrame(tbl, copycols=true)
```

### CODE REVIEW

File: src/simple_reader.jl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    column_names = keys(chunks[1])
+
+    for key in column_names
+        # combine all chunks' key into one column
+        one_column = reduce(vcat, chunk[key] for chunk in chunks)
+        push!(tmp_result_arr, one_column)
+    end
+
+    NamedTuple{column_names}(tmp_result_arr)
```
Comment (tk3369) [inline src/simple_reader.jl:30]: Would this be slow when there are many columns? My use case is probably not very normal but I will probably need to read 10,000 columns.

File: src/simple_reader.jl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    read_parquet(path)
+
+Returns the table contained in the parquet file in an Tables.jl compatible format.
+
+One can easily convert the returned object to any Tables.jl compatible table e.g.
+DataFrames.DataFrame via
+
+```
+using DataFrames
+df = DataFrame(Parquet.read(path))
```
Comment (sa-) [inline src/simple_reader.jl:11]: Shouldn't it be `read_parquet`?
Also, I would add `; copycols=false` to the DataFrame constructor

Comment (xiaodaigh) [inline src/simple_reader.jl:11]:
```suggestion
df = DataFrame(read_parquet(path; copycols=false))
```

File: src/simple_reader.jl (hunk 1 above)
Comment (xiaodaigh) [reply src/simple_reader.jl:30]: What's the best way to do this then without introducing a dependency on DataFrames?

File: src/simple_reader.jl (hunk 2 above)
Comment (xiaodaigh) [reply src/simple_reader.jl]: done

File: src/simple_reader.jl (hunk 1 above)
Comment (xiaodaigh) [reply src/simple_reader.jl]: solved it by returning a Dict instead of NamedTuple

**Comment by xiaodaigh** [comment]
@tanmaykm do u have time to review? I think it's good to go.

**Comment by tanmaykm** [comment]
Sure, will do.
Maybe @quinnj would also like to have a look?

**Comment by tanmaykm** [comment]
This can be a simple API for small files. Larger files will need implementing the Tables.jl interface.
Can merge this in a bit.

**Comment by xiaodaigh** [comment]
What do u mean? This api can read any file right? Anything other than this simple api can go in another PR i think.

Am i missing something here?

**Comment by tanmaykm** [comment]
> Anything other than this simple api can go in another PR

Yes, true.

File: src/simple_reader.jl [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+"""
+function read_parquet(path)
+    parquetfile = Parquet.File(path);
+
+    # read all the chunks
+    chunks = [chunk for chunk in BatchedColumnsCursor(parquetfile)]
+
+    result_dict = Dict()
+
+    column_names = keys(chunks[1])
```
Comment (quinnj) [inline src/simple_reader.jl:22]: We should probably account for "empty" tables here, right? Like if `isempty(chunks) === true`?

File: src/simple_reader.jl [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    for key in column_names
+        # combine all chunks' key into one column
+        one_column = reduce(vcat, chunk[key] for chunk in chunks)
```
Comment (quinnj) [inline src/simple_reader.jl:26]: Oof, this is going to be really expensive if you have really large chunks/files. Ideally, we could use something like `SentinelArrays.ChainedVector` to form a more lazy chain from each chunk's column. This is also the same approach other languages take that I'm familiar with (i.e. C++ arrow/parquet, etc.).

File: src/simple_reader.jl [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        result_dict[key] = one_column
+    end
+
+    result_dict
```
Comment (quinnj) [inline src/simple_reader.jl:30]: I think this is....ok, but there could be advantages to returning a dedicated struct, like implementing the `Tables.partitions` interface to allow compatible sinks to natively know the "partitions" of a chunked parquet file.

**Review by quinnj — COMMENTED** [review]
Sorry for the slow review here, but I've been trying to get caught up on things with all the holidays going on. I left a few comments; happy to help contribute to address some of the things I mentioned.

**Pull request merged**